### PR TITLE
Cleanup and new repository management for add-on store

### LINKED ISSUE #5732: Add-on Store - move repositories to dialog
## The request

The UI for the Add-On Store has always kind of prioritized the wrong information IMO. When you click on a page called "Add-On Store" you expect to see _actual add-ons_, but instead you see a bunch of GitHub repo cards first.

Having just 4 of these cards means you already can't see any add-ons on a mobile device without having to scroll first - not good. And even on desktop it's not great either, as you can see in this example below: 

repos

That is an enormous waste of space for what is essentially nothing more than just URL's. The "Add new repository by URL" card is not as easy to see, it blends in with all the other ones. And as you add more repos the placement of that card is going to shift around the screen rather than being locked in a specific place - so you can never form any muscle memory for where the "add" button is.

It's also inconsistent - these github cards are larger in size and cannot be clicked on, the add-on cards just beneath them are smaller in size and _can_ be clicked on. So two different card behaviors on the very same page, not good.

<hr>

### Proposal

There would maybe be a FAB / floating action button with "Manage Repositories" text. This doesn't necessarily _have_ to be a FAB, it could be some button near the top right of the page or something (maybe to the right of the search field?). 

manage-button

We could figure out how the button would look / be placed, but the more important part of this proposal is moving it to a dialog. When you click the button, it brings up a dialog where you can add or edit the repositories, like so:

repos

This is inspired by both the "Helpers" UI and the custom repos in HACS. This is a much better use of space. A lot of users are going to be happy with the default add-ons and never bother adding any custom repos. Other users might only add maybe one or two repos during their whole time using Home Assistant. 

Adding and removing repos is just not an action you do that often, so it doesn't need to be front and center all the time. This proposal solves that problem and would make the add-on store page look a lot cleaner IMO by putting the focus on the add-ons themselves.

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

- New repository management dialog modeled after the input select helper.
- Changes filter for advanced addons.
- Changed the search to load the same as the integration page.
- Add a link to the profile if advanced mode is not active (similar to /config).

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: closes #5732
- This PR is related to issue:
- Link to documentation pull request:

image
image
image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Would be nice in future PR to show if / how many advances addons are hidden in the missing message.

File: hassio/src/addon-store/hassio-addon-store.ts [hunk 1]
```diff
@@ -12,15 +12,18 @@ import {
   HassioAddonRepository,
   reloadHassioAddons,
 } from "../../../src/data/hassio/addon";
+import "../../../src/components/ha-icon";
 import "../../../src/layouts/loading-screen";
 import "../../../src/layouts/hass-tabs-subpage";
+import { navigate } from "../../../src/common/navigate";
 import { HomeAssistant, Route } from "../../../src/types";
-import "../components/hassio-search-input";
+import "../../../src/common/search/search-input";
```
Comment (bramkragten) [inline hassio/src/addon-store/hassio-addon-store.ts:20]: :tada:

File: src/common/search/search-input.ts [hunk 2]
```diff
@@ -47,14 +50,18 @@ class SearchInput extends LitElement {
         @value-changed=${this._filterInputChanged}
         .noLabelFloat=${this.noLabelFloat}
       >
-        <ha-icon icon="hass:magnify" slot="prefix" class="prefix"></ha-icon>
+        <ha-icon
+          icon="${this.hassio ? "hassio" : "hass"}:magnify"
```
Comment (bramkragten) [inline src/common/search/search-input.ts:54]: No need for this anymore 😆

File: hassio/src/dialogs/repositories/show-dialog-repositories.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+export const showRepositoriesDialog = (
+  element: HTMLElement,
+  dialogParams: HassioRepositoryDialogParams
+): void => {
+  fireEvent(element, "show-dialog", {
+    dialogTag: "dialog-hassio-repositories",
+    dialogImport: () =>
+      import(
+        /* webpackChunkName: "ialog-hassio-repositories" */ "./dialog-hassio-repositories"
```
Comment (bramkragten) [inline hassio/src/dialogs/repositories/show-dialog-repositories.ts:18]:
```suggestion
        /* webpackChunkName: "dialog-hassio-repositories" */ "./dialog-hassio-repositories"
```

File: src/common/search/search-input.ts (hunk 2 above)
Comment (ludeeus) [reply src/common/search/search-input.ts:54]: Noooooooooo, I just learned I should do it this way.... 🤣

File: src/resources/styles.ts [hunk 4]
```diff
@@ -227,6 +227,7 @@ export const haStyleDialog = css`
     --mdc-dialog-max-width: 600px;
     --mdc-dialog-heading-ink-color: var(--primary-text-color);
     --mdc-dialog-content-ink-color: var(--primary-text-color);
+    --mdc-theme-surface: var(--primary-background-color);
```
Comment (bramkragten) [inline src/resources/styles.ts:230]: Why not `card-background-color` like it is now?

File: hassio/src/addon-store/hassio-addon-store.ts [hunk 5]
```diff
@@ -92,28 +95,52 @@ class HassioAddonStore extends LitElement {
...
+                ></search-input>
+              </div>
 
               ${repos}
             `}
+        ${!this.hass.userData?.showAdvanced
+          ? html`
+              <div class="advanced">
+                Missing add-ons? Enable advanced mode on
+                <div class="advanced-link" @click=${this._openProfile}>
```
Comment (bramkragten) [inline hassio/src/addon-store/hassio-addon-store.ts:137]: Make this an `<a>`

**Review by bramkragten — APPROVED** [review]

File: hassio/src/addon-store/hassio-addon-store.ts (hunk 5 above)
Comment (ludeeus) [reply hassio/src/addon-store/hassio-addon-store.ts:137]: The store runs in an iframe, an a would cause it to load the entire site (dual sidebars)

Comment (ludeeus) [reply hassio/src/addon-store/hassio-addon-store.ts:137]: maybe not iframe I do not quite remember, but something.
It was an `<a>` initially, but I had to change that.

Comment (bramkragten) [reply hassio/src/addon-store/hassio-addon-store.ts:137]: you can set the target to `_top` or `_parent`

Comment (ludeeus) [reply hassio/src/addon-store/hassio-addon-store.ts:137]: Thanks! I'll try that

Comment (ludeeus) [reply hassio/src/addon-store/hassio-addon-store.ts]: It worked 🎉

File: src/resources/styles.ts (hunk 4 above)
Comment (ludeeus) [reply src/resources/styles.ts]: Changed.

File: src/resources/styles.ts [hunk 6]
```diff
@@ -227,7 +227,7 @@ export const haStyleDialog = css`
     --mdc-dialog-max-width: 600px;
     --mdc-dialog-heading-ink-color: var(--primary-text-color);
     --mdc-dialog-content-ink-color: var(--primary-text-color);
-    --mdc-theme-surface: var(--primary-background-color);
+    --mdc-theme-surface: var(--paper-card-background-color);
```
Comment (bramkragten) [inline src/resources/styles.ts:230]: No need to put that in here, it is already derived in `styles.ts`

Comment (ludeeus) [reply src/resources/styles.ts:230]: I see, so I should fix it  there instead then, it currently points to `--card-background-color`
Which results in:
image

Comment (ludeeus) [reply src/resources/styles.ts:230]: it's probably the theme I'm using that is incomplete, I'll remove it.

**Pull request merged**